CONTRIBUTING.md

```diff
@@ -4,6 +4,59 @@ Government employees, public and members of the private sector are encouraged to
+2. You have 1 new file added and you may need to make minor changes to syntax or variable names in 25-30 other files. This would be fine to submit as a single PR.
+
+Segmenting a PR is more of an art than a science and PRs typically won't be rejected if they're larger than 20 files. How you segment may be informed by how you choose to branch in git. An example segmentation workflow is provided below.
+
+### Segmentation technique
+
+One common technique used for segmenting PRs is to create a "parent" feature branch off of main. From there, you can create many "child" branches from the parent. A PR is submitted each time you want to merge a child into its parent. A child branch will represent one logical division of the work and, as mentioned above, will contain no more than 20 changed files (depending on complexity of changes).
+
+![PR Segmentation Diagram](docs/pr-segmentation.png)
+
+With this method of segmentation, you don't need to worry about committing incomplete or broken features to main. When all child branches have been completed for the feature, the parent is then merged back into main in a final PR. This final review can contain your CHANGELOG.md entry. In the title or description of your PR, please indicate that all work has already been reviewed.
+
+## Merge Techniques
+
+Due to how PRs are commonly segmented in this repository, and due to its active nature, rebasing is discouraged as a way to merge changes. Traditional merge commits are preferred as they preserve history and make it easier to resolve merge conflicts.
+
+## Guidelines for reviewers
+
+Conditions for whether to Approve or Request Changes for a PR can come with a lot of nuance. We try to air on the side of Approve unless there are obvious changes needed. Below is a list of reasons to "stop" a PR. Items with a star next to them are subject to interpretation by the review. Some items may prompt to discussion between the reviewer and the submitting developer so that choices can be justified.
+
+Examples of when to Request Changes
+- Logging functions are left in that aren't required or needed later
+- Commented-code is left in that isn't needed later
+- Widespread incorrect syntax use (using snake case for many variables in JS when it's not required)
+- Significant amount of repeition in code that can be condensed without much issue (code isn't DRY) *
+- Unnecessary function calls, overly verbose code *
+
+## Common Components/Utilities
+
+- `@transational`: Database method decorator. If there is an exception during execution, the entire DB session will be safely rolled back to a point in time just before the decorated function was called. If not, the session will be saved, unless autocommit is set to False. This helps replace most session management boilerplate.
+- `authorization.check_auth`: Checks a user for one or more roles and for the correct engagement.
```

Comment:
These are really helpful to have documented. I will definitely be adding to this section in the future :)

Reply:
Awesome! I'm hoping that having a reference list of common utilities could speed up development in the future